src/ripple/app/misc/impl/ValidatorList.cpp

```diff
@@ -170,6 +170,12 @@ ValidatorList::load (
             return false;
         }
 
+        if (publicKeyType(*id)!= KeyType::secp256k1)
+        {
+            // Trusted ed25519 keys will be added to the manifest cache
+            continue;
+        }
```

Comment:
`validation_create` currently only generates secp256k1. However, you're the second person to point out that this prevents us from allowing other validation key types.
I have one alternative to checking key types. We could initially put everything under `[validators]` on the permanent keys list. Then we would identify master keys when we see them included in a received manifest.